When I use that with the #23353 branch I get with `sage -p gfan`:
```
...
src/field_rationalfunctions2.o: In function `FieldElementRationalFunction2::norm
alize(bool)':
/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunction
s2.cpp:228: undefined reference to `polynomialGCD(Polynomial const&, Polynomial 
const&)'
src/field_rationalfunctions2.o: In function `FieldElementRationalFunction2::madd(FieldElementImplementation const&, FieldElementImplementation const&)':
/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunctions2.cpp:412: undefined reference to `polynomialGCD(Polynomial const&, Polynomial const&)'
/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunctions2.cpp:413: undefined reference to `polynomialGCD(Polynomial const&, Polynomial const&)'
src/field_rationalfunctions2.o: In function `FieldElementRationalFunction2::operator*=(FieldElementImplementation const&)':
/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunctions2.cpp:334: undefined reference to `polynomialGCD(Polynomial const&, Polynomial const&)'
/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunctions2.cpp:335: undefined reference to `polynomialGCD(Polynomial const&, Polynomial const&)'
src/field_rationalfunctions2.o:/home/ralf/sage/local/var/tmp/sage/build/gfan-0.6/src/src/field_rationalfunctions2.cpp:469: more undefined references to `polynomialGCD(Polynomial const&, Polynomial const&)' follow
collect2: error: ld returned 1 exit status
```

--- a/Makefile
+++ b/Makefile
@@ -85,6 +85,8 @@ OPTFLAGS    =  -DGMPRATIONAL -Wuninitialized -fno-omit-frame-pointer -O2	 #-O3 -
 CFLAGS	  := $(CFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
 CXXFLAGS  := $(CXXFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
 
+SINGULAR_OBJECTS = src/polynomialgcd.o
+
 CATSOBJECTS =	lp_cdd.o \
 		parser.o \
 		field.o \
@@ -358,7 +360,6 @@ src/symmetrictraversal.o: src/symmetrictraversal.cpp
 
 OBJECTS = 	$(addprefix src/,$(SOPLEX_OBJECTS)) \
 		$(SINGULAR_OBJECTS) \
-		$(SAGE_OBJECTS) \
 		$(addprefix src/,$(CATSOBJECTS)) \
 		$(addprefix src/,$(APPOBJECTS)) \
 		src/symmetrictraversal.o

